boxoffice/models/invoice.py

```diff
@@ -0,0 +1,93 @@
+    invoiced_at = db.Column(db.DateTime, nullable=True)
+    street_address = db.Column(db.Unicode(255), nullable=True)
+    city = db.Column(db.Unicode(255), nullable=True)
+    state = db.Column(db.Unicode(255), nullable=True)
+    # India specific: this is the state short code. Eg: KA for Karnataka
+    state_code = db.Column(db.Unicode(4), nullable=True)
+    # ISO country code
+    country_code = db.Column(db.Unicode(2), nullable=True)
+    postcode = db.Column(db.Unicode(8), nullable=True)
+    # GSTIN in the case of India
+    buyer_taxid = db.Column(db.Unicode(255), nullable=True)
+    seller_taxid = db.Column(db.Unicode(255), nullable=True)
+
+    customer_order_id = db.Column(None, db.ForeignKey('customer_order.id'), nullable=False, index=True)
+    order = db.relationship('Order', backref=db.backref('invoices', cascade='all, delete-orphan'))
+
+    organization_id = db.Column(None, db.ForeignKey('organization.id'), nullable=False)
+    organization = db.relationship('Organization', backref=db.backref('invoices', cascade='all, delete-orphan', lazy='dynamic'))
+
+    def __init__(self, *args, **kwargs):
+        organization = kwargs.get('organization')
+        if not organization:
+            raise ValueError(u"Invoice MUST be initialized with an organization")
+        self.invoice_no = get_latest_invoice_no(organization) + 1
+        self.invoiced_at = func.utcnow()
+        super(Invoice, self).__init__(*args, **kwargs)
+
+
+@event.listens_for(Invoice, 'before_update')
+@event.listens_for(Invoice, 'before_insert')
+def validate_invoice_organization(mapper, connection, target):
+    if target.organization != target.order.organization:
+        raise ValueError(u"Invoice MUST be associated with the same organization as its order")
+
+
+class InvoiceLineItem(UuidMixin, BaseMixin, db.Model):
+    __tablename__ = 'invoice_line_item'
+    __uuid_primary_key__ = True
+    __table_args__ = (db.UniqueConstraint('invoice_id', 'seq'),)
+
+    seq = db.Column(db.Integer, nullable=False)
+    item_title = db.Column(db.Unicode(255), nullable=False)
+    quantity = db.Column(db.SmallInteger, nullable=False, default=1)
+    # In India, this will be GST
+    tax_type = db.Column(db.Unicode(255), nullable=False)
+    gst_type = db.Column(db.Unicode(7), nullable=False)
+    discount_title = db.Column(db.Unicode(255), nullable=False)
+
+    currency = db.Column(db.Unicode(3), nullable=False)
+    base_amount = db.Column(db.Numeric, default=Decimal(0), nullable=False)
+    discounted_amount = db.Column(db.Numeric, default=Decimal(0), nullable=False)
+    # Total amount includes tax
+    total_amount = db.Column(db.Numeric, default=Decimal(0), nullable=False)
+
+    cgst_tax_rate = db.Column(db.Integer, nullable=True, default=0)
+    sgst_tax_rate = db.Column(db.Integer, nullable=True, default=0)
+    igst_tax_rate = db.Column(db.Integer, nullable=True, default=0)
+    gst_compensation_cess = db.Column(db.SmallInteger, nullable=True, default=0)
```

Comment:
> What about UTGST? Does it need to be separate, or is it equivalent to SGST (in filings)? 

They are equivalent.

> If this is the rate stored in each line item, what is the actual value? 

The actual value is computed when required.

> Considering it will not always round-off to two digits when you calculate from the base rate, shouldn't you be storing the actual rounded-off value?

I chose not to store actual values, since calculating the tax amount based on a constant tax rate and a constant base rate will always result in the same value.

I'm going to remove the `InvoiceLineItem` model for now and add it back in the next release after iterating on the model design.